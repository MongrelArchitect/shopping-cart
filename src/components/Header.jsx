import React from 'react';
import { Link } from 'react-router-dom';

export default function Header() {
  return (
    <div className="header">
      <span>FakeShop</span>
      <ul>
        <li>
          <Link to="/">Home</Link>
        </li>
        <li>
          <Link to="/shop">Shop</Link>
        </li>
        <li>
          <Link to="/cart">Cart</Link>
        </li>
      </ul>
    </div>
  );
}
