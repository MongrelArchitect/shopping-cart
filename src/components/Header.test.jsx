import React from 'react';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import Header from './Header';

describe('Header component', () => {
  it('Renders correctly with zero cart items', () => {
    const { container } = render(
      <BrowserRouter>
        <Header cart={{}} />
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });

  it('Shows correct number of items in cart', () => {
    const fakeCart = {
      1: 7,
      2: 1,
      4: 3,
    };

    render(
      <BrowserRouter>
        <Header cart={fakeCart} />
      </BrowserRouter>,
    );

    const cartItems = screen.getByText(/\(\d* item(s)\)/);
    expect(cartItems.textContent).toMatch('(11 items)');
  });

  it('Style links according to router path', () => {
    render(
      <MemoryRouter initialEntries={['/shop']}>
        <Header cart={{}} />
      </MemoryRouter>,
    );

    const shopLink = screen.getByText('Shop');
    const homeLink = screen.getByText('Home');
    const cartLink = screen.getByText('Cart');
    expect(shopLink.classList.contains('active')).toBe(true);
    expect(homeLink.classList.contains('active')).toBe(false);
    expect(cartLink.classList.contains('active')).toBe(false);
  });
});
