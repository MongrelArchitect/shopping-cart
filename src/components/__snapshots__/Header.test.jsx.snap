// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Renders correctly with zero cart items 1`] = `
<div>
  <div
    class="header"
  >
    <h1>
      FakeShop
    </h1>
    <ul>
      <li>
        <a
          class="active"
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/shop"
        >
          Shop
        </a>
      </li>
      <li>
        <a
          href="/cart"
        >
          Cart
        </a>
        <span>
           
          (
          0
           
          item
          s
          )
        </span>
      </li>
    </ul>
  </div>
</div>
`;
